### DISCUSSION TITLE
QST:Using geopandas in flask, the program ends directly

### DISCUSSION THREAD

**Discussion by Roey1996** [geopandas/geopandas#2728] [Q&A]
- [x] I have searched the [geopandas] tag on [StackOverflow](https://stackoverflow.com/questions/tagged/geopandas) and [GIS StackExchange](https://gis.stackexchange.com/questions/tagged/geopandas) for similar questions.

- [ ] I have asked my usage related question on [StackOverflow](https://stackoverflow.com) or [GIS StackExhange](https://gis.stackexchange.com).

---   my question

A simple code read shapefile, if i use it in python env, **it can show out the result**;

like this
![image](https://user-images.githubusercontent.com/33124375/186096328-3085d449-615a-4cfc-a5e9-35a4af94c3c4.png)

**but** when i put it into **flask** program,when i use flask route , the program ends directly,

Here is my flask use code
`
@DataTransformation.route('/TransShape2Jsonfile',methods=['GET','POST'])
def TransShape2Jsonfile():
    s2g.TransShapefile2JsonFile()
    return 'Data Trans Over'
`

![image](https://user-images.githubusercontent.com/33124375/186096552-dfcd1efd-16eb-4e91-8c9c-edb6ea8fa38d.png)

`gpd.read_file(FilePath, encoding='gbk')`

After testing, it is found that it is **read_ The file()** function cannot be run. When the program runs here, it will terminate directly

```

**Comment by aaryan-verma** [comment]
I want to work on this issue. Pls assign it to me.

**Comment by martinfleis** [comment]
@aaryan-verma it is a question. If you know what is happening feel free to answer it. Unless it uncovers a bug on geopandas side, there's nothing to do code-wise.

@1996HZP do you have a traceback? I believe that a server should print one. How did you install geopandas and its dependencies?
